storage/internal/queue.go

```diff
@@ -103,6 +126,30 @@ func (q *Queue) Add(ctx context.Context, e *tessera.Entry) tessera.IndexFuture {
 	return qi.f
 }
 
+// Close ensures that all work in the queue is flushed and processed. After this is done,
+// the Queue will be locked for any more calls to Add.
+func (q *Queue) Close(ctx context.Context) error {
+	q.mu.Lock()
+	defer q.mu.Unlock()
+	q.closed = true
+
+	// Flush and close the buffer, which will trigger workers
+	// to process the entries.
+	if err := q.buf.Flush(); err != nil {
+		return err
+	}
```

Comment:
I think you can drop the call to `Flush` as `Close` will flush before marking the buffer closed: https://pkg.go.dev/github.com/globocom/go-buffer#Buffer.Close